core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -682,11 +677,93 @@ object ConditionalScriptPubKey extends ScriptFactory[ConditionalScriptPubKey] {
 
-    (headIsOpIf && endsWithEndIf && validConditionalTree, opElseIndex)
+    (headIsConditional && endsWithEndIf && validConditionalTree, opElseIndex)
+  }
+
+  def apply(
+      conditional: ConditionalOperation,
+      trueSPK: RawScriptPubKey,
+      falseSPK: RawScriptPubKey): ConditionalScriptPubKey = {
+    conditional match {
+      case OP_IF    => IfConditionalScriptPubKey(trueSPK, falseSPK)
+      case OP_NOTIF => NotIfConditionalScriptPubKey(falseSPK, trueSPK)
+    }
+  }
+}
+
+sealed trait IfConditionalScriptPubKey extends ConditionalScriptPubKey {
+  override def conditional: ConditionalOperation = OP_IF
+}
+
+object IfConditionalScriptPubKey
+    extends ScriptFactory[IfConditionalScriptPubKey] {
+  private case class IfConditionalScriptPubKeyImpl(
+      override val asm: Vector[ScriptToken])
+      extends IfConditionalScriptPubKey {
+    override def toString: String =
+      s"IfConditionalScriptPubKey($trueSPK, $falseSPK)"
+  }
+
+  override def fromAsm(asm: Seq[ScriptToken]): IfConditionalScriptPubKey = {
+    buildScript(
+      asm = asm.toVector,
+      constructor = IfConditionalScriptPubKeyImpl.apply,
+      invariant = isIfConditionalScriptPubKey,
+      errorMsg = "Given asm was not a IfConditionalScriptPubKey, got: " + asm
+    )
+  }
+
+  def apply(
+      trueSPK: RawScriptPubKey,
+      falseSPK: RawScriptPubKey): IfConditionalScriptPubKey = {
+    val asm = Vector(OP_IF) ++ trueSPK.asm ++ Vector(OP_ELSE) ++ falseSPK.asm ++ Vector(
+      OP_ENDIF)
+
+    fromAsm(asm)
+  }
+
+  def isIfConditionalScriptPubKey(asm: Seq[ScriptToken]): Boolean = {
+    ConditionalScriptPubKey
+      .isConditionalScriptPubKeyWithElseIndex(asm, OP_IF)
+      ._1
+  }
+}
+
+sealed trait NotIfConditionalScriptPubKey extends ConditionalScriptPubKey {
+  override def conditional: ConditionalOperation = OP_NOTIF
+}
+
+object NotIfConditionalScriptPubKey
+    extends ScriptFactory[NotIfConditionalScriptPubKey] {
+  private case class NotIfConditionalScriptPubKeyImpl(
+      override val asm: Vector[ScriptToken])
+      extends NotIfConditionalScriptPubKey {
+    override def toString: String =
+      s"NotIfConditionalScriptPubKey($falseSPK, $trueSPK)"
+  }
+
+  override def fromAsm(asm: Seq[ScriptToken]): NotIfConditionalScriptPubKey = {
+    buildScript(
+      asm = asm.toVector,
+      constructor = NotIfConditionalScriptPubKeyImpl.apply,
+      invariant = isNotIfConditionalScriptPubKey,
+      errorMsg = "Given asm was not a NotIfConditionalScriptPubKey, got: " + asm
+    )
+  }
+
+  def apply(
+      falseSPK: RawScriptPubKey,
+      trueSPK: RawScriptPubKey): NotIfConditionalScriptPubKey = {
+    val asm = Vector(OP_NOTIF) ++ falseSPK.asm ++ Vector(OP_ELSE) ++ trueSPK.asm ++ Vector(
+      OP_ENDIF)
+
+    fromAsm(asm)
   }
 
-  def isConditionalScriptPubKey(asm: Seq[ScriptToken]): Boolean = {
-    isConditionalScriptPubKeyWithElseIndex(asm)._1
+  def isNotIfConditionalScriptPubKey(asm: Seq[ScriptToken]): Boolean = {
```

Comment:
not that this should be done this PR, but we should probably just start calling these `isValid()` and add them as a def in `ScriptFactory`

Reply:
I think that could be nice :)

Reply:
Definitely won't clutter it here though